File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int st =0,end = nums.size()-1,mid=0,ans=INT_MAX;
        while(st<=end){
            mid = st + (end-st)/2;
            if(nums[st] <= nums[mid]){
                ans = min(ans,nums[st]);
                st = mid+1;
            }
            else{
                ans = min(ans,nums[mid]);
                end = mid-1;
            }
        }
        return ans;
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] arr) {
        int mini = Integer.MAX_VALUE;
         int idx = 0;
         int st =0,end = arr.length -1;
        while(st<= end){
         int mid = st + (end - st)/2;

         if(arr[st] <= arr[mid]){
            if(mini > arr[st]){
                mini = arr[st];
                
            }
            st = mid+1;
         }
         else{
             if(mini > arr[mid]){
                mini = arr[mid];
                
             }
             end = mid-1;
         }
        }
        return mini;
    }
}